src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerConfigBuilder.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerConfig.Scope;
+
+/**
+ * Builds a {@link CircuitBreakerConfig} instance using builder pattern.
+ */
+public final class CircuitBreakerConfigBuilder {
+
+    private static final double DEFAULT_FAILURE_RATE_THRESHOLD = 0.8;
+
+    private static final Scope DEFAULT_SCOPE = Scope.SERVICE;
+
+    private static final ExceptionFilter DEFAULT_FAILURE_FILTER = cause -> true;
+
+    private static final long DEFAULT_MINIMUM_REQUEST_THRESHOLD = 10;
+
+    private static final Duration DEFAULT_TRIAL_REQUEST_INTERVAL = Duration.ofSeconds(3);
+
+    private static final Duration DEFAULT_CIRCUIT_OPEN_WINDOW = Duration.ofSeconds(10);
+
+    private static final Duration DEFAULT_COUNTER_SLIDING_WINDOW = Duration.ofSeconds(20);
+
+    private static final Duration DEFAULT_COUNTER_UPDATE_INTERVAL = Duration.ofSeconds(1);
+
+    private final String remoteServiceName;
+
+    private double failureRateThreshold = DEFAULT_FAILURE_RATE_THRESHOLD;
+
+    private Scope scope = DEFAULT_SCOPE;
+
+    private ExceptionFilter exceptionFilter = DEFAULT_FAILURE_FILTER;
+
+    private long minimumRequestThreshold = DEFAULT_MINIMUM_REQUEST_THRESHOLD;
+
+    private Duration trialRequestInterval = DEFAULT_TRIAL_REQUEST_INTERVAL;
+
+    private Duration circuitOpenWindow = DEFAULT_CIRCUIT_OPEN_WINDOW;
+
+    private Duration counterSlidingWindow = DEFAULT_COUNTER_SLIDING_WINDOW;
+
+    private Duration counterUpdateInterval = DEFAULT_COUNTER_UPDATE_INTERVAL;
+
+    /**
+     * Creates a new {@link CircuitBreakerConfigBuilder} with the specified name of remote service.
+     *
+     * @param remoteServiceName The remote service name to be logged
+     */
+    public CircuitBreakerConfigBuilder(String remoteServiceName) {
```

Comment:
Could we make `remoteServiceName` optional? For example, if a user did not specify the service name, the circuit breaker could use the URI of the endpoint automatically.